app.rb

```diff
@@ -0,0 +1,133 @@
+require_relative 'book'
+require_relative 'classroom'
+require_relative 'person'
+require_relative 'rental'
+require_relative 'student'
+require_relative 'teacher'
+
+class App
+  attr_accessor :people, :books, :rentals
+
+  def initialize
+    @books = []
+    @people = []
+    @rentals = []
+  end
+
+  def list_all_books
+    puts '#=> No books added yet' if books.empty?
+    books.each do |book|
+      puts "#=> Title: #{book.title} Author: #{book.author}"
+    end
+    puts people.length
+  end
+
+  def list_all_people
+    puts '#=> No person added yet' if books.empty?
+
+    people.each do |person|
+      puts "#=> [#{person.class}] ID: #{person.id} Name: #{person.name} Age: #{person.age}"
+    end
+    puts people.length
+  end
+
+  def create_teacher
+    print "Teacher's specialization: "
+    specialization = gets.chomp
+    print 'Age: '
+    age = gets.chomp
+    print 'Name: '
+    name = gets.chomp
+    teacher = Teacher.new(specialization, name, age)
+    people << teacher
+    puts ''
+    puts '#=> Teacher created successfully'
+  end
+
+  def create_student
+    print 'Age: '
+    age = Integer(gets.chomp)
+    print 'Name: '
+    name = gets.chomp
+    print 'Parent permission? [y/n]: '
+    parent_permission = gets.chomp.downcase
+
+    case parent_permission
+    when 'n'
+      people << Student.new(nil, age, name, false)
+    when 'y'
+      people << Student.new(nil, age, name)
+    else
+      '#=> Invalid option, please try again'
+    end
+
+    puts ''
+    puts '#=> Student created successfully'
+  end
```

Comment:
- [ ] Great job implementing suggested changes from previous reviewers. As mentioned by @anagudelogu in the previous review, there's a wrong passsing of arguments when creating a student. This happens when you're about to specify the parent_paermission status of the student. I got below error message which the cause of that has been explained by @anagudelogu in his review. Kindly look into it and fix.

### Error Message
![wrong-args-student](https://github.com/aima98/OOP-School-Library/assets/26597432/c34ca3c0-0b6a-4111-bb3b-45a93a3cd9aa)